### PR TITLE
feat(feg_hello): counters for agw-orc8r-feg latency

### PR DESCRIPTION
**Pull request by uri200**
Signed-off-by: Oriol Batalla <obatalla@fb.com>

## Summary

Added some counters to see if we can catch latency between elements.

Note the only true value is `total time` which is taken at the client sending request and client receiving request. The other counters are best effort and may depend on how well synchronized are AGW, orc8r and Feg with time servers. 

## Test Plan

Tested using federated local setup on mac

```
(python) vagrant@magma-dev-focal:~$ feg_hello_cli.py m 0
- Request:
 greeting: "m"

- Response: m

- Stats:
  * Total time:     61 ms
  * Aproximate path (ms):
    ┌─────┐─>     15 ─>┌───────┐->     16 ─>┌─────┐
    │ AGW │            │ Orc8r │            │ Feg │
    └─────┘<─ 10     <─└───────┘<─ 20     <─└─────┘
```

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: feg/protos/hello.proto [hunk 1]
```diff
@@ -40,13 +42,24 @@ message HelloRequest {
...
+  google.protobuf.Timestamp agw_to_feg_relay_timestamp = 3;
 }
 
 message HelloReply {
   string greeting = 1;
+  ResponseTimestamps timestamps = 2;
+}
+
+message ResponseTimestamps {
+  // timestamp oming from HelloRequest
```
Comment (emakeev) [inline feg/protos/hello.proto]: coming ...

**Review by emakeev — COMMENTED** [review]
Here is another hello client, you may want to update: https://github.com/magma/magma/blob/master/feg/gateway/tools/hello_cli/main.go

**Comment by uri200** [comment]
> Here is another hello client, you may want to update: https://github.com/magma/magma/blob/master/feg/gateway/tools/hello_cli/main.go

Oh that's true. Will update it too.

But if this one is fine i would like to land before tomorrow (Tuesday) so we can have, at least, the feg abd orc8r  part in place so we can test  it in real deployments like teravm

**Review by themarwhal — APPROVED** [review]

File: lte/gateway/python/scripts/feg_hello_cli.py [hunk 2]
```diff
@@ -22,10 +23,35 @@
...
 @cloud_grpc_wrapper
 def echo(client, args):
+    start_time = datetime.datetime.now()
     req = HelloRequest(greeting=args.msg, grpc_err_code=args.err_code)
-    print("request:\n", req)
+    print("- Request:\n", req)
     resp = client.SayHello(req)
-    print("resp: ", resp)
+    end_time = datetime.datetime.now()
+    print(f'- Response: {resp.greeting}')
```
Comment (ardzoht) [inline lte/gateway/python/scripts/feg_hello_cli.py:31]: should this be inside the `if resp` condition?

File: lte/gateway/python/scripts/feg_hello_cli.py [hunk 3]
```diff
@@ -45,6 +71,24 @@ def create_parser():
     return parser
 
 
+def to_ms_string(seconds, left=False):
+    """
+    Convert seconds into a sting of milliseconds. If the string is smaller
```
Comment (ardzoht) [inline lte/gateway/python/scripts/feg_hello_cli.py]:
```suggestion
    Convert seconds into a string of milliseconds. If the string is smaller
```

File: lte/gateway/python/scripts/feg_hello_cli.py (hunk 2 above)
Comment (uri200) [reply lte/gateway/python/scripts/feg_hello_cli.py:31]: good catch,  I think i forgot to remove that

**Review by ardzoht — APPROVED** [review]
🚀

**Review by emakeev — APPROVED** [review]

**Comment by emakeev** [comment]
> But if this one is fine i would like to land before tomorrow (Tuesday) so we can have, at least, the feg abd orc8r part in place so we can test it in real deployments like teravm

Sure, it definitely can go into a separate PR

**Review by themarwhal — APPROVED** [review]

**Pull request merged**